repo/saxpy/saxpy.cc

```diff
@@ -110,28 +119,73 @@ int main(int argc, char** argv) {
 
     int rank, size;
     MPI_Init(&argc, &argv);
-    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
-    MPI_Comm_size(MPI_COMM_WORLD, &size);
+    MPI_Comm_rank(comm, &rank);
+    MPI_Comm_size(comm, &size);
+
+    #ifdef USE_CALIPER
+      /*-----------------------------------------------------------
+       * Set Caliper and Adiak metadata
+       *----------------------------------------------------------*/
+      adiak_init(&comm);
+      adiak_user();
+      adiak_uid();
+      adiak_launchdate();
+      adiak_executable();
+      adiak_executablepath();
+      adiak_libraries();
+      adiak_cmdline();
+      adiak_hostname();
+      adiak_clustername();
```

Comment:
If you're using Adiak v0.4.0+ you can use `adiak_collect_all()` to get all the Adiak built-in metadata